### ISSUE TITLE
ROCm 2.7 - work_group_scan_exclusive_add not working

### ISSUE THREAD

**Issue by Moading** [RadeonOpenCompute/ROCm#866]
Hi,

the kernel below does not compile on ROCm 2.7

__kernel void scan(__global int * a, __global int * b)
{
	size_t local_id=get_local_id(0);
	int value=a[local_id];
	int prefix_sum = work_group_scan_exclusive_add(value);
	b[local_id]=prefix_sum+value;
}

The build log contains the following error:

build log:
error -11, status=-2, log: 
: error: undefined hidden symbol: __oclc_wavefrontsize64
>>> referenced by /tmp/t_2545_118-9d1dc7.o:(scan)
>>> referenced by /tmp/t_2545_118-9d1dc7.o:(scan)
Error: Creating the executable from LLVM IRs failed.
end of build log.

If it matters, the OS is Ubuntu 18.04.3 LTS up to date.

Rock on!

**Comment by b-sumner** [comment]
I assume you have a new navi board and are compiling for it?  Are you sure you're using exclusively only ROCm 2.7 bits?  Nothing in your path or LD_LIBRARY_PATH that could be providing a different OpenCL runtime or compiler library?

**Comment by Moading** [comment]
I'm using a Radeon Pro Duo, the same machine as in the other issues I opened.
LD_LIBRARY_PATH points to MPI and boost only.

Should I post other information?

**Comment by b-sumner** [comment]
Something is strange then.  If you're not using navi, then a different compiler is used that knows nothing about navi and uses libraries that know nothing about wavefrontsize64.  

Do you have environment variables set that have AMD and/or GPU in their names?

**Comment by Moading** [comment]
Both

env | grep AMA
env | grep GPU

return nothing.

**Comment by b-sumner** [comment]
I don't believe that by default for gfx803, you should be using the compiler apparently being used and seeing this issue.  We're looking into this.

**Comment by Moading** [comment]
Great, happy looking!

**Comment by b-sumner** [comment]
I had forgotten that the ROCm OpenCL runtime only contains the one compiler, while the LinuxPro and Windows OpenCL runtime contains two compilers.  So that question is answered.

Also, we believe that if the 2.7 release is built from source, the problem you've hit will not appear.

We need to verify your observation and try to understand what happened if we do.

**Comment by fweiglhofer** [comment]
I wanted to add, that I've encountered the same issue with the related `work_group_scan_inclusive_add` function.

**Comment by Moading** [comment]
Are you running ROCm 2.7.1? The output of 'dkms status' will tell.

**Comment by fweiglhofer** [comment]
`dkms status` gives me 2.7-22 as version number.

**Comment by Moading** [comment]
@b-summer
Hi, were you able to reproduce the problem?

**Comment by sree314** [comment]
I'm having a similar issue:
```
: error: undefined hidden symbol: __oclc_wavefrontsize64
>>> referenced by /tmp/t_69964_33-4e8da3.o:(reduce_float_kernel_builtin)
>>> referenced by /tmp/t_69964_33-4e8da3.o:(reduce_float_kernel_builtin)
Error: Creating the executable from LLVM IRs failed.

```
Ubuntu 14.0.3 LTS, here are the packages:
```
ii  rocm-clang-ocl                             0.4.0.35-rocm-rel-2.7-22-7ce124f             amd64        OpenCL compilation with clang compiler.
ii  rocm-cmake                                 0.3.0.93-rocm-rel-2.7-22-b727cef             amd64        rocm-cmake built using CMake
ii  rocm-dev                                   2.7.22                                       amd64        Radeon Open Compute (ROCm) Runtime software stack
ii  rocm-device-libs                           0.0.1                                        amd64        Radeon Open Compute - device libraries
ii  rocm-dkms                                  2.7.22                                       amd64        Radeon Open Compute (ROCm) Runtime software stack
ii  rocm-opencl                                1.2.0-2019080925                             amd64        OpenCL/ROCm
ii  rocm-opencl-dev                            1.2.0-2019080925                             amd64        OpenCL/ROCm
ii  rocm-smi                                   1.0.0-167-ge14d23e                           amd64        System Management Interface for ROCm
ii  rocm-utils                                 2.7.22                                       amd64        Radeon Open Compute (ROCm) Runtime software stack
ii  rocminfo                                   1.0.0                                        amd64        Radeon Open Compute (ROCm) Runtime rocminfo tool
```

**Comment by Moading** [comment]
howdy @sree314,
you are running ROcm 2.7.1 could you please install ROCm 2.7.2 and test if it works? I can no test on my machines as I am traveling...

**Comment by Moading** [comment]
The problem persists in ROCm 2.7.2

**Comment by sree314** [comment]
@Moading , yep, just got the packages now, and the error persists.

Using Device: Vega 10 XT [Radeon RX Vega 64]
Driver Version: 2949.0 (HSA1.1,LC)

(though I haven't rebooted the machine, which I'll do later today).

```
ii  rocm-clang-ocl                             0.4.0.35-rocm-rel-2.7-27-7ce124f             amd64        OpenCL compilation with clang compiler.
ii  rocm-cmake                                 0.3.0.93-rocm-rel-2.7-27-b727cef             amd64        rocm-cmake built using CMake
ii  rocm-dev                                   2.7.27                                       amd64        Radeon Open Compute (ROCm) Runtime software stack
ii  rocm-device-libs                           0.0.1                                        amd64        Radeon Open Compute - device libraries
ii  rocm-dkms                                  2.7.27                                       amd64        Radeon Open Compute (ROCm) Runtime software stack
ii  rocm-opencl                                1.2.0-2019082828                             amd64        OpenCL/ROCm
ii  rocm-opencl-dev                            1.2.0-2019082828                             amd64        OpenCL/ROCm
ii  rocm-smi                                   1.0.0-167-ge14d23e                           amd64        System Management Interface for ROCm
ii  rocm-utils                                 2.7.27                                       amd64        Radeon Open Compute (ROCm) Runtime software stack
ii  rocminfo                                   1.0.0                                        amd64        Radeon Open Compute (ROCm) Runtime rocminfo tool

```

**Comment by Moading** [comment]
The problem is fixed in ROCm 2.8, thanks!
